src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -53,37 +71,150 @@ public AuthorisedCaseSearchOperation(
-        JsonNode data = objectMapperService.convertObjectToJsonNode(caseDetails.getData());
-        JsonNode filteredData = accessControlService.filterCaseFieldsByAccess(data, caseType.getCaseFields(), getUserRoles(), CAN_READ);
-        caseDetails.setData(objectMapperService.convertJsonNodeToMap(filteredData));
+    private void filterCaseDataByAclAccess(CaseType authorisedCaseType, CaseDetails caseDetails) {
+        JsonNode caseData = caseDataToJsonNode(caseDetails);
+        JsonNode accessFilteredData = accessControlService.filterCaseFieldsByAccess(caseData, authorisedCaseType.getCaseFields(), getUserRoles(), CAN_READ);
+        caseDetails.setData(jsonNodeToCaseData(accessFilteredData));
     }
 
-    private void filterCaseFieldsBySecurityClassification(CaseDetails caseDetails) {
+    private void filterCaseDataBySecurityClassification(CaseDetails caseDetails) {
         classificationService.applyClassification(caseDetails);
     }
 
+    /**
+     * Filters the case data to the aliases that were passed in the _source filter of the search request. For e.g. if the case data is
+     * "case_data": {
+     *   "PersonFirstName" : "J",
+     *   "PersonLastName": "Baker",
+     *   "PersonAddress": {
+     *     "city": "London",
+     *     "postcode": "W4"
+     *   }
+     * }
+     *
+     * and the source filter is
+     * "_source": ["alias.lastName", "alias.postcode"]
+     * where alias.lastName = case_data.PersonLastName and alias.postcode = case_data.PersonAddress.postcode
+     *
+     * the case data will be filtered and transformed to
+     * "case_data": {
+     *   "lastName": "Baker",
+     *   "postcode": "W4",
+     * }
+     *
+     * If no source filter was passed then this will remove case data and return only metadata.
+     */
+    private void filterCaseDataForMultiCaseTypeSearch(CrossCaseTypeSearchRequest searchRequest, CaseType authorisedCaseType, CaseDetails caseDetails) {
+        if (searchRequest.isMultiCaseTypeSearch() && caseDetails.getData() != null) {
+            JsonNode caseData = caseDataToJsonNode(caseDetails);
+            String caseDataJson = caseData.toString();
+            JsonNode filteredMultiCaseTypeSearchData = objectMapperService.createEmptyJsonNode();
+
+            authorisedCaseType.getSearchAliasFields()
+                .stream()
+                .filter(searchRequest::hasAliasField)
+                .forEach(searchAliasField -> findCaseFieldPathInCaseData(authorisedCaseType, caseDataJson, searchAliasField.getCaseFieldPath())
+                    .filter(not(JsonNode::isMissingNode))
+                    .ifPresent(jsonNode -> ((ObjectNode) filteredMultiCaseTypeSearchData).set(searchAliasField.getId(), jsonNode)));
+
+            caseDetails.setData(jsonNodeToCaseData(filteredMultiCaseTypeSearchData));
+        }
+    }
+
+    private Optional<JsonNode> findCaseFieldPathInCaseData(CaseType caseType, String caseDataJson, String path) {
+        try {
```

Comment:
Using jsonpath parser in place of JsonNode at method. This has rich support and fixes issue with collection field lookups